### DISCUSSION TITLE
Can keycloak be configured to connect to multiple FreeIPA replicas for User Federation?

### DISCUSSION THREAD

**Discussion by keestux** [keycloak/keycloak#34904] [Q&A] [answered]
I have a FreeIPA system with multiple replicas. They are redundant, which is the whole point of having more than one.
I understand that I can add a "User federation" with the "Kerberos provider". In the configuration I have to select one of the FreeIPA replicas for the connection. How can I configure Keycloak to try more than one FreeIPA server? Is that at all possible?

**Comment by darius-m** [comment, marked as answer]
Do you require Kerberos in particular? For LDAP, at least, you can add multiple servers using an URL like `ldaps://server-1:636 ldaps://server-2:636` (separated by spaces). The only downside is that Keycloak will always connect to the first available server (I think this is also what happens in LDAP libraries for other programming languages, so it might be expected).

**Reply by keestux** [reply]
Ah, OK. That sounds like a good alternative, FreeIPA does LDAP too. Thanks.

I'll have to see how bad the timeout is when the first replica is offline.

**Reply by f1-outsourcing** [reply]
> For LDAP, at least, you can add multiple servers using an URL like `ldaps://server-1:636 ldaps://server-2:636` (separated by spaces). The only downside is that Keycloak will always connect to the first available server (I think this is also what happens in LDAP libraries for other programming languages, so it might be expected).

But if one goes down during operation, it will take the next?

**Reply by darius-m** [reply]
That's what I meant with "the first available" - if the first one goes down, it should go to the second, but as soon as the first one comes back up, it will only connect to that one. Unless something has changed since I last checked, that behavior should be pretty reliable.
